src/index.js

```diff
@@ -0,0 +1,221 @@
+    rule: {
+      type: RANGE_TYPE,
+      ipMinimum: normalizedIp1,
+      ipMaximum: normalizedIp2,
+    },
+  };
+}
+
+function processSubnet(rule) {
+  const match = rule.match(/^(.*)\/(\d{1,3})$/);
+
+  if (!match) {
+    return {
+      isValid: false,
+      match: false,
+    };
+  }
+
+  const [, ip, mask] = match;
+  const maskNumber = parseInt(mask, 10);
+
+  if (!rangeCheck.isIP(ip)) {
+    return {
+      isValid: false,
+      match: true,
+      reason: 'IP is invalid',
+    };
+  }
+
+  const normalizedIp = rangeCheck.displayIP(ip);
+  const ipVersion = rangeCheck.ver(normalizedIp);
+
+  if (ipVersion === 4 && (maskNumber < 0 || mask > 32)) {
+    return {
+      isValid: false,
+      match: true,
+      reason: 'Mask must be between 0 and 32',
+    };
+  } else if (maskNumber < 0 || mask > 128) {
+    return {
+      isValid: false,
+      match: true,
+      reason: 'Mask must be between 0 and 128',
+    };
+  }
+
+  return {
+    isValid: true,
+    rule: {
+      type: SUBNET_TYPE,
+      range: `${normalizedIp}/${mask}`,
+    },
+  };
+}
+
+function stringToRuleObject(ruleString) {
+  const trimmedRule = ruleString.trim().replace(/ /g, '');
+
+  let result = processIP(trimmedRule);
+  if (result.isValid || result.match) {
+    return result;
+  }
+
+  result = processRange(trimmedRule);
+  if (result.isValid || result.match) {
+    return result;
+  }
+
+  result = processSubnet(trimmedRule);
+  if (result.isValid || result.match) {
+    return result;
+  }
+
+  return { isValid: false, reason: 'Badly constructed rule' };
+}
+
+function isIpMatchesRule(ip, rule) {
+  const normalizedIp = rangeCheck.displayIP(ip);
+
+  if (rule.type === IP_TYPE) {
+    return normalizedIp === rule.ip;
+  } else if (rule.type === RANGE_TYPE) {
+    const ipVersion = rangeCheck.ver(normalizedIp);
+    const rangeVersion = rangeCheck.ver(rule.ipMinimum);
+
+    if (ipVersion !== rangeVersion) {
+      return false;
+    }
+
+    if (ipVersion === 4) {
+      const ipValueMinimum = ipV4ToInt(rule.ipMinimum);
+      const ipValueMaximum = ipV4ToInt(rule.ipMaximum);
+      const ipValue = ipV4ToInt(normalizedIp);
+
+      return ipValue >= ipValueMinimum && ipValue <= ipValueMaximum;
+    }
+
+    const ipValueMinimum = ipV6ToBigInteger(rule.ipMinimum);
+    const ipValueMaximum = ipV6ToBigInteger(rule.ipMaximum);
+    const ipValue = ipV6ToBigInteger(normalizedIp);
+
+    return ipValue.compareTo(ipValueMinimum) >= 0 && ipValue.compareTo(ipValueMaximum) <= 0;
+  } else if (rule.type === SUBNET_TYPE) {
+    return rangeCheck.inRange(normalizedIp, rule.range);
+  }
+
+  throw new Error('Invalid rule type');
+}
+
+function contain(list, ip) {
+  return list.some((rule) => {
+    const ruleObject = stringToRuleObject(rule.value);
+
+    if (ruleObject.isValid === false) {
+      throw new Error(`Invalid rule: ${ruleObject.reason}`, rule);
+    }
+
+    if (isIpMatchesRule(ip, ruleObject.rule)) {
+      return true;
+    }
+
+    return false;
```

Comment:
`return isIpMatchesRule(ip, ruleObject.rule);` is a simpler code.(1 line of code instead of 5)

Reply:
```javascript
let matchFound = false;
if (ruleObject.isValid) {
  matchFound = isIpMatchesRule(ip, ruleObject.rule);
}

return matchFound;
```